src/www/system_advanced_admin.php

```diff
@@ -362,6 +382,40 @@
      });
      $(".proto").change();
 
+     // Show/hide options for WebGUI management access.
+     $("#webgui_managementaccess").change(function(){
+         var mgmt_type = 'webgui_managementaccess_' + $(this).val();
+         $(".webgui_managementaccess_choices").hide();
+         $("."+mgmt_type).show();
+     });
+     $("#webgui_managementaccess").change();
+
+     $('#webgui_managementaccess').change(function () {
+         if ($('#webgui_managementaccess option:selected').val() == 'interfaces') {
+             $.webgui_managementaccess_warned = 0;
+         } else if ($.webgui_managementaccess_warned != 1) {
+             $.webgui_managementaccess_warned = 1;
+             BootstrapDialog.confirm({
+                 title: '<?= html_safe(gettext('Warning!')) ?>',
+                 message: '<?= html_safe(gettext('Changing the management access configuration of the web GUI may ' .
+                     'prevent you from accessing this page if you continue. It is recommended to keep ' .
+                     'this set to the default unless you know what you are doing.')) ?>',
+                 type: BootstrapDialog.TYPE_WARNING,
+                 btnOKClass: 'btn-warning',
+                 btnOKLabel: '<?= html_safe(gettext('I know what I am doing')) ?>',
+                 btnCancelLabel: '<?= html_safe(gettext('Use the default')) ?>',
+                 callback: function(result) {
+                     if (!result) {
+                         $('#webgui_managementaccess option:selected').prop('selected', false);
+                         $('#webgui_managementaccess').selectpicker('refresh');
```

Comment:
```suggestion
                         $('#webgui_managementaccess').selectpicker('refresh');
                         $("#webgui_managementaccess").change();
```

To correctly display currently relevant input field. At least in Firefox the `selectpicker('refresh')` does not trigger the _change_ event.

Reply:
Good catch, thanks.